src/WebJobs.Extensions.CosmosDB/CosmosDBSqlResolutionPolicy.cs

```diff
@@ -36,18 +35,44 @@ public string TemplateBind(PropertyInfo propInfo, Attribute resolvedAttribute, B
             SqlParameterCollection paramCollection = new SqlParameterCollection();
 
             // also build up a dictionary replacing '{token}' with '@token' 
-            IDictionary<string, string> replacements = new Dictionary<string, string>();
+            IDictionary<string, object> replacements = new Dictionary<string, object>();
             foreach (var token in bindingTemplate.ParameterNames.Distinct())
             {
-                string sqlToken = $"@{token}";
-                paramCollection.Add(new SqlParameter(sqlToken, bindingData[token]));
-                replacements.Add(token, sqlToken);
+                object tokenObject = bindingData[token];
+                if (tokenObject is string tokenString)
+                {
+                    string sqlToken = GetSqlParameterName(token);
+                    paramCollection.Add(new SqlParameter(sqlToken, tokenString));
+                    replacements.Add(token, sqlToken);
+                }
+                else if (tokenObject is IDictionary<string, string> tokenDictionary)
+                {
+                    IDictionary<string, string> tokens = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+                    foreach (var item in tokenDictionary)
+                    {
+                        string fullToken = $"{token}.{item.Key}";
+                        bool tokenInUse = bindingTemplate.Pattern.Contains(fullToken);
+                        if (tokenInUse)
+                        {
+                            string sqlToken = GetSqlParameterName(fullToken);
+                            paramCollection.Add(new SqlParameter(sqlToken, item.Value));
+                            tokens.Add(item.Key, sqlToken);
```

Comment:
This logic seems duplicated from the other above, could they be encapsulated in a helper method to avoid some future break by an edit that forgets the update both?

Reply:
Refactored out the duplicate logic. Hope I didn't make it worse. 😄